toolkit/tools/pkg/imagecustomizerlib/extractpartitions.go

```diff
@@ -79,11 +118,165 @@ func compressWithZstd(partitionRawFilepath string) (partitionFilepath string, er
 		return "", fmt.Errorf("failed to compress %s with zstd:\n%w", partitionRawFilepath, err)
 	}
 
-	// Remove raw file since output partition format is raw-zst.
-	err = os.Remove(partitionRawFilepath)
+	return partitionRawFilepath + ".zst", nil
+}
+
+// Prepend a skippable frame with the metadata payload to the specified partition file.
+func addSkippableFrame(partitionFilepath string, magicNumber uint32, frameSize uint32, payload [4]uint32) (err error) {
+	// Read existing data from the partition file.
+	existingData, err := os.ReadFile(partitionFilepath)
+	if err != nil {
+		return fmt.Errorf("failed to read partition file %s:\n%w", partitionFilepath, err)
+	}
+
+	// Create a skippable frame.
+	skippableFrame := createSkippableFrame(magicNumber, frameSize, payload)
+
+	// Combine the skippable frame and existing data.
+	newData := append(skippableFrame, existingData...)
+
+	// Write the combined data back to the partition file.
+	err = os.WriteFile(partitionFilepath, newData, 0644)
 	if err != nil {
-		return "", fmt.Errorf("failed to remove raw file %s:\n%w", partitionRawFilepath, err)
+		return fmt.Errorf("failed to write skippable frame to partition file %s:\n%w", partitionFilepath, err)
 	}
 
-	return partitionRawFilepath + ".zst", nil
+	return nil
+}
+
+// Creates a skippable frame.
+func createSkippableFrame(magicNumber uint32, frameSize uint32, payload [4]uint32) (skippableFrame []byte) {
+	// Calculate the length of the byte array
+	lengthOfByteArray := 4 + 4 + (4 * len(payload))
+	// Define the Skippable frame
+	skippableFrame = make([]byte, lengthOfByteArray)
+	// Magic_Number
+	binary.LittleEndian.PutUint32(skippableFrame, magicNumber)
+	// Frame_Size
+	binary.LittleEndian.PutUint32(skippableFrame[4:8], frameSize)
+	// User_Data
+	binary.LittleEndian.PutUint32(skippableFrame[8:12], payload[0])
```

Comment:
Converting the 128-bits into u32s and then back into bytes seems a little pointless. You should be able to use the copy function.

```golang
var payload [16]byte
...
copy(skippableFrame[8:24], payload[:])
```

Reply:
Just updated it to use this logic.